raiden/network/proxies/secret_registry.py

```diff
@@ -65,8 +64,7 @@ def register_secret(self, secret: typing.Secret):
             contract=pex(self.address),
         )
 
-        transaction_hash = estimate_and_transact(
-            self.proxy,
+        transaction_hash = self.proxy.transact(
```

Comment:
`self.proxy.functions.registerSecret(secret).transact()`

Reply:
no, this must be done via the proxy (because of local signatures)

Reply:
Ah, Ok

Reply:
So now, all transactions must use `self.proxy.transact`, but calls to query variables can be made through `self.proxy.functions`. Correct?

Reply:
@loredanacirstea for now, yes. But once the `web3` middleware signatures are merged & we decide to switch, we can use standard `web3.eth.functions.f().transact()`.

Reply:
Good, because 2 ways of doing transact & call is quite confusing at the moment.